### PR TITLE
Finding Linked List is Palindrome or Not

### PR DESCRIPTION
**Pull request by sumitgsh**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
YES
### Description
Linked List Palindrome check using two pointers approach
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#2969

### CODE REVIEW

**Comment by sumitgsh** [comment]
@HarshCasper @siddharth25pandey @ankitaggarwal23 please review my PR

File: C/ds/FindLinkedListPalindrome.c [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        Linked_List[4]==arr[2]
+        Linked_List[5]==arr[1]
+        ...
+        ...
+        Palindrome(If all elements from array are matched)
+
+*/
+
+#include <malloc.h>
```
Comment (ankitaggarwal23) [inline C/ds/FindLinkedListPalindrome.c]: Use `#include<bits/stdc++.h>` instead of `<malloc.h>`

File: C/ds/FindLinkedListPalindrome.c [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  status = find_Palindrome(size);
+
+  if (status) {
+    printf("\nThe above Linked List is Palindrome");
+  } else {
+    printf("\nThe above Linked List is Not Palindrome");
+  }
+
+  return 0;
+}
```
Comment (ankitaggarwal23) [inline C/ds/FindLinkedListPalindrome.c:118]: - Align your code properly. 
- Use proper code Format 
- Give one tab spaces

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C/ds/FindLinkedListPalindrome.c (hunk 2 above)
Comment (sumitgsh) [reply C/ds/FindLinkedListPalindrome.c:118]: @ankitaggarwal23 I used clang-format as a formatter.
And I used one tab space on every line after that I tried to format the code using clang formatted as mentioned.After that only I pushed the code. Can u be more specific like which line is disorienting?

File: C/ds/FindLinkedListPalindrome.c (hunk 1 above)
Comment (sumitgsh) [reply C/ds/FindLinkedListPalindrome.c]: @ankitaggarwal23  I could have used <bits/stdc++.h> but only compiler like vscode supports that..unlike Devc++,Turbo C doesn't have any such Standard library. Even though u want me to change,I can do that  I don't have any issue.

Comment (sumitgsh) [reply C/ds/FindLinkedListPalindrome.c]: Please suggest should I need to change since my compiler(DevC++) is also not supporting it.

Comment (HarshCasper) [reply C/ds/FindLinkedListPalindrome.c]: Please make the change. We will try to follow a standard

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by sumitgsh** [comment]
@HarshCasper @ankitaggarwal23 @iamrajiv  can u please review the changes and merge

File: C/ds/FindLinkedListPalindrome.c [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+Note:when no of elements are odd in Linked List skip the middle element
+Ex- Linked_List={1,2,3,4,3,2,1},size=7
+        Storing the first half in an array
+        arr=[1,2,3]
+
+        Linked_List[4]==arr[2]
+        Linked_List[5]==arr[1]
+        ...
+        ...
+        Palindrome(If all elements from array are matched)
```
Comment (HarshCasper) [inline C/ds/FindLinkedListPalindrome.c]: Remove this

File: C/ds/FindLinkedListPalindrome.c [hunk 4]
```diff
@@ -0,0 +1,152 @@
+/*C program to find whether a linked list is palindrome or not
+Using two pointers approach i.e slow(moves at x speed)pointer and fast(moves at
+2x speed)pointer By storing first half of the linked list in an array and then
+comparing each element from the other half of the linked list
+with the elements of array in reverse order till it get exhausted.
```
Comment (HarshCasper) [inline C/ds/FindLinkedListPalindrome.c]: Lots of grammatical and stylistic mistakes. Fix this. Don't assume things from a technical perspective. Talk about it from a layman's point of view

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C/ds/FindLinkedListPalindrome.c (hunk 4 above)
Comment (sumitgsh) [reply C/ds/FindLinkedListPalindrome.c]: Thanks for the feedback @HarshCasper can u please review the changes requested.

File: C/ds/FindLinkedListPalindrome.c (hunk 3 above)
Comment (sumitgsh) [reply C/ds/FindLinkedListPalindrome.c]: Removed can u please verify once and merge @HarshCasper @iamrajiv

File: C/ds/FindLinkedListPalindrome.c [hunk 5]
```diff
@@ -0,0 +1,145 @@
+/* C program to find whether a linked list is palindrome or not.
+We need to detect whether the  front half of the list is the reverse of the second half.
+By reversing the front half of the list and for that stack can be used.
+
+We know the size of the linked list,we can iterate onto the first half of the elements in a for loop ,
+pushing each element onto a stack.Now ,we iterate the rest of linked list.At each step we compare
+the node to the top of the stack.If we complete the iteration without finding the difference,then the 
+linked list is palindrome
+
```
Comment (HarshCasper) [inline C/ds/FindLinkedListPalindrome.c]: Run this through Grammarly and fix the grammar.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C/ds/FindLinkedListPalindrome.c (hunk 5 above)
Comment (sumitgsh) [reply C/ds/FindLinkedListPalindrome.c]: @HarshCasper Fixed through Grammarly and made the required changes, can u please review

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**